mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala

```diff
@@ -562,4 +562,20 @@ trait HasBlockSize extends Params {
   /** @group expertGetParam */
   final def getBlockSize: Int = $(blockSize)
 }
+
+/**
+ * Trait for shared param k. This trait may be changed or
+ * removed between minor versions.
+ */
+trait HasK extends Params {
+
```

Comment:
Add a simple test in https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/ml/param/shared/SharedParamsSuite.scala ?

like 
```
class Obj(override val uid: String) extends Params with HasOutputCol with HasK {
...

assert(obj.getOrDefault(obj.k) == 2)

```

Reply:
@keypointt 
We normally test the parameters (default value, getter and setter) in each of the individual algorithm. We have existing tests that cover default value of k, getK, setK in all the clustering algorithms.

Reply:
cool, I'm all good then. thanks :)